src/CoreWCF.RabbitMQ.Client/src/CoreWCF/ServiceModel/Channels/RabbitMqChannelFactory.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System.Linq;
+using System;
+using System.ServiceModel.Channels;
+using System.ServiceModel.Security;
+using System.Threading.Tasks;
+using CoreWCF.Runtime;
+using System.IdentityModel.Selectors;
+using ClientCredentials = System.ServiceModel.Description.ClientCredentials;
+
+namespace CoreWCF.ServiceModel.Channels
+{
+    internal class RabbitMqChannelFactory : ChannelFactoryBase<IOutputChannel>
+    {
+        private BufferManager _bufferManager;
+        private MessageEncoderFactory _messageEncoderFactory;
+        private RabbitMqTransportBindingElement _transport;
+        private SecurityCredentialsManager _channelCredentials;
+        private SecurityTokenManager _securityTokenManager;
+
+        internal RabbitMqChannelFactory(RabbitMqTransportBindingElement transport, BindingContext context)
+            : base(context.Binding)
+        {
+            _transport = transport;
+            _bufferManager = BufferManager.CreateBufferManager(transport.MaxBufferPoolSize, int.MaxValue);
+            InitializeSecurityTokenManager(context);
+
+            var messageEncoderBindingElement = context.BindingParameters.Find<MessageEncodingBindingElement>();
+            _messageEncoderFactory = messageEncoderBindingElement == null
+                ? RabbitMqDefaults.DefaultMessageEncoderFactory
+                : messageEncoderBindingElement.CreateMessageEncoderFactory();
+        }
+
+        private void InitializeSecurityTokenManager(BindingContext context)
+        {
+            _channelCredentials = context.BindingParameters.OfType<SecurityCredentialsManager>().FirstOrDefault();
+            _channelCredentials ??= new ClientCredentials();
+            _securityTokenManager = _channelCredentials.CreateSecurityTokenManager();
+        }
+
+        public RabbitMqTransportBindingElement Transport => _transport;
+
```

Comment:
You don't need to define a field. If you have a getter only property, you are allowed to assign it in a constructor.

Reply:
The same applies to BufferManager and MessageEncoderFactory